From a04762c1177c62075f9a106815d63fe1e4a7a883 Mon Sep 17 00:00:00 2001
From: Lubomir Rintel <lkundrak@v3.sk>
Date: Fri, 4 Feb 2011 18:00:38 +0100
Subject: [PATCH] Fix build with GCC 4.6

Add cstddef headers for ptrdiff_h and size_t. Add cstddef also befor uses
of sigc++ to work around it also using ptrdiff_h w/o proper header.

Newer gcc no longer passws --export-dynamic to linker transparently,
make it do so via -Wl.
---
 src/2geom/utils.h                           |    1 +
 src/Makefile.am                             |    2 +-
 src/application/editor.h                    |    1 +
 src/arc-context.h                           |    1 +
 src/box3d-context.h                         |    1 +
 src/conn-avoid-ref.h                        |    1 +
 src/connection-pool.h                       |    1 +
 src/connector-context.h                     |    1 +
 src/desktop.h                               |    1 +
 src/dialogs/object-attributes.cpp           |    1 +
 src/display/canvas-temporary-item.h         |    1 +
 src/document-private.h                      |    1 +
 src/document-subset.h                       |    1 +
 src/document.h                              |    1 +
 src/draw-context.h                          |    1 +
 src/extension/timer.h                       |    1 +
 src/flood-context.h                         |    1 +
 src/gc-alloc.h                              |    1 +
 src/gc-finalized.h                          |    1 +
 src/gc.cpp                                  |    1 +
 src/gradient-context.h                      |    1 +
 src/gradient-drag.h                         |    1 +
 src/knot.h                                  |    1 +
 src/libavoid/geomtypes.h                    |    1 +
 src/live_effects/lpeobject-reference.h      |    1 +
 src/live_effects/parameter/path-reference.h |    1 +
 src/live_effects/parameter/path.h           |    1 +
 src/message-stack.h                         |    1 +
 src/object-hierarchy.h                      |    1 +
 src/persp3d-reference.h                     |    1 +
 src/rect-context.h                          |    1 +
 src/selcue.h                                |    1 +
 src/selection-describer.h                   |    1 +
 src/selection.h                             |    1 +
 src/seltrans.h                              |    1 +
 src/sp-conn-end-pair.h                      |    1 +
 src/sp-conn-end.h                           |    1 +
 src/sp-gradient.h                           |    1 +
 src/sp-object.h                             |    1 +
 src/sp-offset.h                             |    1 +
 src/sp-pattern.h                            |    1 +
 src/sp-shape.h                              |    1 +
 src/sp-switch.h                             |    1 +
 src/sp-text.h                               |    1 +
 src/sp-tref-reference.h                     |    1 +
 src/sp-use-reference.h                      |    1 +
 src/sp-use.h                                |    1 +
 src/spiral-context.h                        |    1 +
 src/star-context.h                          |    1 +
 src/style.h                                 |    1 +
 src/text-context.h                          |    1 +
 src/ui/dialog/desktop-tracker.h             |    1 +
 src/ui/dialog/document-metadata.h           |    1 +
 src/ui/dialog/document-properties.h         |    1 +
 src/ui/dialog/inkscape-preferences.h        |    1 +
 src/ui/dialog/undo-history.cpp              |    1 +
 src/ui/tool/control-point-selection.h       |    1 +
 src/ui/tool/control-point.h                 |    1 +
 src/ui/tool/manipulator.h                   |    1 +
 src/ui/tool/multi-path-manipulator.h        |    1 +
 src/ui/tool/node-tool.h                     |    1 +
 src/ui/tool/node.h                          |    1 +
 src/ui/view/view.h                          |    1 +
 src/ui/widget/color-picker.h                |    1 +
 src/ui/widget/page-sizer.h                  |    1 +
 src/ui/widget/preferences-widget.h          |    1 +
 src/ui/widget/selected-style.h              |    1 +
 src/ui/widget/style-subject.h               |    1 +
 src/uri-references.h                        |    1 +
 src/util/forward-pointer-iterator.h         |    1 +
 src/util/share.h                            |    1 +
 src/util/unordered-containers.h             |    2 ++
 src/widgets/desktop-widget.h                |    1 +
 src/widgets/gradient-image.h                |    1 +
 src/widgets/gradient-vector.h               |    1 +
 src/widgets/sp-attribute-widget.h           |    1 +
 src/widgets/sp-color-notebook.cpp           |    1 +
 src/xml/helper-observer.h                   |    1 +
 78 files changed, 79 insertions(+), 1 deletions(-)

diff --git a/src/2geom/utils.h b/src/2geom/utils.h
index 5ab1917..c505ff4 100644
--- a/src/2geom/utils.h
+++ b/src/2geom/utils.h
@@ -33,6 +33,7 @@
  *
  */
 
+#include <cstddef>
 #include <cmath>
 #include <vector>
 
diff --git a/src/Makefile.am b/src/Makefile.am
index 03b58c6..6ee7e46 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -205,7 +205,7 @@ libinkscape_a_SOURCES = $(ink_common_sources)
 
 inkscape_SOURCES += main.cpp $(win32_sources)
 inkscape_LDADD = $(all_libs)
-inkscape_LDFLAGS = --export-dynamic $(kdeldflags) $(mwindows)
+inkscape_LDFLAGS = -Wl,--export-dynamic $(kdeldflags) $(mwindows)
 
 inkview_SOURCES += inkview.cpp $(win32_sources)
 inkview_LDADD = $(all_libs)
diff --git a/src/application/editor.h b/src/application/editor.h
index 4545022..b77e0ad 100644
--- a/src/application/editor.h
+++ b/src/application/editor.h
@@ -15,6 +15,7 @@
 #ifndef INKSCAPE_APPLICATION_EDITOR_H
 #define INKSCAPE_APPLICATION_EDITOR_H
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <glib/gslist.h>
 #include <glibmm/ustring.h>
diff --git a/src/arc-context.h b/src/arc-context.h
index 1e18222..3ed4478 100644
--- a/src/arc-context.h
+++ b/src/arc-context.h
@@ -15,6 +15,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 #include <2geom/point.h>
diff --git a/src/box3d-context.h b/src/box3d-context.h
index 8bafa75..69b4a05 100644
--- a/src/box3d-context.h
+++ b/src/box3d-context.h
@@ -15,6 +15,7 @@
  * Released under GNU GPL
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "event-context.h"
 #include "proj_pt.h"
diff --git a/src/conn-avoid-ref.h b/src/conn-avoid-ref.h
index 5dff8dd..653b1d7 100644
--- a/src/conn-avoid-ref.h
+++ b/src/conn-avoid-ref.h
@@ -14,6 +14,7 @@
  */
 
 #include <glib/gslist.h>
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 struct SPDesktop;
diff --git a/src/connection-pool.h b/src/connection-pool.h
index 67bcf00..4aa44e3 100644
--- a/src/connection-pool.h
+++ b/src/connection-pool.h
@@ -3,6 +3,7 @@
 
 #include <glib-object.h>
 #include <gtkmm.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 namespace Inkscape
diff --git a/src/connector-context.h b/src/connector-context.h
index bd3805e..d977162 100644
--- a/src/connector-context.h
+++ b/src/connector-context.h
@@ -12,6 +12,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <sigc++/connection.h>
 #include "event-context.h"
diff --git a/src/desktop.h b/src/desktop.h
index af2473b..5ca4aa5 100644
--- a/src/desktop.h
+++ b/src/desktop.h
@@ -27,6 +27,7 @@
 
 #include <gdk/gdkevents.h>
 #include <gtk/gtktypeutils.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 #include <2geom/matrix.h>
diff --git a/src/dialogs/object-attributes.cpp b/src/dialogs/object-attributes.cpp
index 320840f..6530869 100644
--- a/src/dialogs/object-attributes.cpp
+++ b/src/dialogs/object-attributes.cpp
@@ -13,6 +13,7 @@
 #include <glibmm/i18n.h>
 #include <string>
 #include <cstring>
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include <sigc++/functors/ptr_fun.h>
 #include <sigc++/adaptors/bind.h>
diff --git a/src/display/canvas-temporary-item.h b/src/display/canvas-temporary-item.h
index 5077fb5..61e096a 100644
--- a/src/display/canvas-temporary-item.h
+++ b/src/display/canvas-temporary-item.h
@@ -14,6 +14,7 @@
 
 #include "display/display-forward.h"
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 namespace Inkscape {
diff --git a/src/document-private.h b/src/document-private.h
index fa47542..cc9872d 100644
--- a/src/document-private.h
+++ b/src/document-private.h
@@ -14,6 +14,7 @@
  */
 
 #include <map>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "xml/event-fns.h"
 #include "sp-defs.h"
diff --git a/src/document-subset.h b/src/document-subset.h
index e424a28..f76d708 100644
--- a/src/document-subset.h
+++ b/src/document-subset.h
@@ -10,6 +10,7 @@
 #ifndef SEEN_INKSCAPE_DOCUMENT_SUBSET_H
 #define SEEN_INKSCAPE_DOCUMENT_SUBSET_H
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include <sigc++/functors/slot.h>
 
diff --git a/src/document.h b/src/document.h
index 5810b53..5500549 100644
--- a/src/document.h
+++ b/src/document.h
@@ -17,6 +17,7 @@
 
 #include <glib-object.h>
 #include <gtk/gtksignal.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <sigc++/class_slot.h>
 
diff --git a/src/draw-context.h b/src/draw-context.h
index 30ebaa6..1d42a71 100644
--- a/src/draw-context.h
+++ b/src/draw-context.h
@@ -14,6 +14,7 @@
  * Released under GNU GPL
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "event-context.h"
 #include <forward.h>
diff --git a/src/extension/timer.h b/src/extension/timer.h
index ebae62b..33b9829 100644
--- a/src/extension/timer.h
+++ b/src/extension/timer.h
@@ -13,6 +13,7 @@
 #ifndef INKSCAPE_EXTENSION_TIMER_H__
 #define INKSCAPE_EXTENSION_TIMER_H__
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <glibmm/timeval.h>
 #include "extension-forward.h"
diff --git a/src/flood-context.h b/src/flood-context.h
index af537de..d9da960 100644
--- a/src/flood-context.h
+++ b/src/flood-context.h
@@ -11,6 +11,7 @@
  * Released under GNU GPL
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <gtk/gtk.h>
 #include "event-context.h"
diff --git a/src/gc-alloc.h b/src/gc-alloc.h
index 83811c0..bcd6d6a 100644
--- a/src/gc-alloc.h
+++ b/src/gc-alloc.h
@@ -16,6 +16,7 @@
 #define SEEN_INKSCAPE_GC_ALLOC_H
 
 #include <limits>
+#include <cstddef>
 #include "gc-core.h"
 
 namespace Inkscape {
diff --git a/src/gc-finalized.h b/src/gc-finalized.h
index cf47cb0..036740b 100644
--- a/src/gc-finalized.h
+++ b/src/gc-finalized.h
@@ -17,6 +17,7 @@
 #define SEEN_INKSCAPE_GC_FINALIZED_H
 
 #include <new>
+#include <cstddef>
 #include "gc-core.h"
 
 namespace Inkscape {
diff --git a/src/gc.cpp b/src/gc.cpp
index ee988af..8622059 100644
--- a/src/gc.cpp
+++ b/src/gc.cpp
@@ -16,6 +16,7 @@
 #include <glib/gmessages.h>
 #include <sigc++/functors/ptr_fun.h>
 #include <glibmm/main.h>
+#include <cstddef>
 
 namespace Inkscape {
 namespace GC {
diff --git a/src/gradient-context.h b/src/gradient-context.h
index 6f8a804..8c2a5f7 100644
--- a/src/gradient-context.h
+++ b/src/gradient-context.h
@@ -15,6 +15,7 @@
  * Released under GNU GPL
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "event-context.h"
 
diff --git a/src/gradient-drag.h b/src/gradient-drag.h
index 8cbe9f3..40ab065 100644
--- a/src/gradient-drag.h
+++ b/src/gradient-drag.h
@@ -15,6 +15,7 @@
  */
 
 #include <glib/gslist.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <vector>
 
diff --git a/src/knot.h b/src/knot.h
index 351c7f7..a651956 100644
--- a/src/knot.h
+++ b/src/knot.h
@@ -20,6 +20,7 @@
 #include "forward.h"
 #include <2geom/point.h>
 #include "knot-enums.h"
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 class SPKnot;
diff --git a/src/libavoid/geomtypes.h b/src/libavoid/geomtypes.h
index ced53e6..61963c4 100644
--- a/src/libavoid/geomtypes.h
+++ b/src/libavoid/geomtypes.h
@@ -29,6 +29,7 @@
 #ifndef AVOID_GEOMTYPES_H
 #define AVOID_GEOMTYPES_H
 
+#include <cstddef>
 #include <vector>
 #include <utility>
 
diff --git a/src/live_effects/lpeobject-reference.h b/src/live_effects/lpeobject-reference.h
index 48d7eb3..8d2b406 100644
--- a/src/live_effects/lpeobject-reference.h
+++ b/src/live_effects/lpeobject-reference.h
@@ -11,6 +11,7 @@
 
 #include <forward.h>
 #include <uri-references.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 namespace Inkscape {
diff --git a/src/live_effects/parameter/path-reference.h b/src/live_effects/parameter/path-reference.h
index 67508e5..739c447 100644
--- a/src/live_effects/parameter/path-reference.h
+++ b/src/live_effects/parameter/path-reference.h
@@ -11,6 +11,7 @@
 
 #include <forward.h>
 #include <uri-references.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 class Path;
diff --git a/src/live_effects/parameter/path.h b/src/live_effects/parameter/path.h
index 23ad514..8fe9382 100644
--- a/src/live_effects/parameter/path.h
+++ b/src/live_effects/parameter/path.h
@@ -16,6 +16,7 @@
 
 #include "live_effects/parameter/parameter.h"
 #include "live_effects/parameter/path-reference.h"
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 namespace Inkscape {
diff --git a/src/message-stack.h b/src/message-stack.h
index 24ec2d5..33b077d 100644
--- a/src/message-stack.h
+++ b/src/message-stack.h
@@ -14,6 +14,7 @@
 #ifndef SEEN_INKSCAPE_MESSAGE_STACK_H
 #define SEEN_INKSCAPE_MESSAGE_STACK_H
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <glib.h>
 #include <stdarg.h>
diff --git a/src/object-hierarchy.h b/src/object-hierarchy.h
index e5f44b4..c127a48 100644
--- a/src/object-hierarchy.h
+++ b/src/object-hierarchy.h
@@ -14,6 +14,7 @@
 
 #include <exception>
 #include <list>
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include <sigc++/signal.h>
 #include <glib/gmessages.h>
diff --git a/src/persp3d-reference.h b/src/persp3d-reference.h
index 7c2ce31..c5f2857 100644
--- a/src/persp3d-reference.h
+++ b/src/persp3d-reference.h
@@ -11,6 +11,7 @@
  */
 
 #include "uri-references.h"
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "persp3d.h"
 
diff --git a/src/rect-context.h b/src/rect-context.h
index 0445338..54f790c 100644
--- a/src/rect-context.h
+++ b/src/rect-context.h
@@ -14,6 +14,7 @@
  * Released under GNU GPL
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "event-context.h"
 #include "libnr/nr-point.h"
diff --git a/src/selcue.h b/src/selcue.h
index c9266ac..0869a59 100644
--- a/src/selcue.h
+++ b/src/selcue.h
@@ -14,6 +14,7 @@
  */
 
 #include <list>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 class SPDesktop;
diff --git a/src/selection-describer.h b/src/selection-describer.h
index 4b0e3d8..10a613e 100644
--- a/src/selection-describer.h
+++ b/src/selection-describer.h
@@ -12,6 +12,7 @@
 #ifndef SEEN_INKSCAPE_SELECTION_DESCRIPTION_HANDLER_H
 #define SEEN_INKSCAPE_SELECTION_DESCRIPTION_HANDLER_H
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "message-context.h"
 
diff --git a/src/selection.h b/src/selection.h
index b5a511e..ae19316 100644
--- a/src/selection.h
+++ b/src/selection.h
@@ -19,6 +19,7 @@
 #include <vector>
 #include <map>
 #include <list>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 //#include "libnr/nr-rect.h"
diff --git a/src/seltrans.h b/src/seltrans.h
index 0183683..11652e1 100644
--- a/src/seltrans.h
+++ b/src/seltrans.h
@@ -15,6 +15,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <2geom/point.h>
 #include <2geom/matrix.h>
diff --git a/src/sp-conn-end-pair.h b/src/sp-conn-end-pair.h
index 3b011ed..bc5fa12 100644
--- a/src/sp-conn-end-pair.h
+++ b/src/sp-conn-end-pair.h
@@ -15,6 +15,7 @@
 
 #include "forward.h"
 #include "libnr/nr-point.h"
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include <sigc++/functors/slot.h>
 #include <sigc++/signal.h>
diff --git a/src/sp-conn-end.h b/src/sp-conn-end.h
index 16a611e..052e8dd 100644
--- a/src/sp-conn-end.h
+++ b/src/sp-conn-end.h
@@ -2,6 +2,7 @@
 #define SEEN_SP_CONN_END
 
 #include <glib/gtypes.h>
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 #include "sp-use-reference.h"
diff --git a/src/sp-gradient.h b/src/sp-gradient.h
index 97ea78f..f0d7f64 100644
--- a/src/sp-gradient.h
+++ b/src/sp-gradient.h
@@ -25,6 +25,7 @@
 #include "sp-gradient-units.h"
 #include "sp-gradient-vector.h"
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 struct SPGradientReference;
diff --git a/src/sp-object.h b/src/sp-object.h
index 6330916..ebb4b6a 100644
--- a/src/sp-object.h
+++ b/src/sp-object.h
@@ -56,6 +56,7 @@
 
 
 #include <glib-object.h>
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include <sigc++/functors/slot.h>
 #include <sigc++/signal.h>
diff --git a/src/sp-offset.h b/src/sp-offset.h
index caecdb1..304ffb0 100644
--- a/src/sp-offset.h
+++ b/src/sp-offset.h
@@ -16,6 +16,7 @@
 
 #include "sp-shape.h"
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 #define SP_TYPE_OFFSET            (sp_offset_get_type ())
diff --git a/src/sp-pattern.h b/src/sp-pattern.h
index f15285e..9c60524 100644
--- a/src/sp-pattern.h
+++ b/src/sp-pattern.h
@@ -32,6 +32,7 @@ class SPPatternClass;
 #include "sp-paint-server.h"
 #include "uri-references.h"
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 class SPPatternReference : public Inkscape::URIReference {
diff --git a/src/sp-shape.h b/src/sp-shape.h
index 4b1ded2..58515ee 100644
--- a/src/sp-shape.h
+++ b/src/sp-shape.h
@@ -19,6 +19,7 @@
 #include "sp-marker-loc.h"
 #include <2geom/forward.h>
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 #define SP_TYPE_SHAPE (sp_shape_get_type ())
diff --git a/src/sp-switch.h b/src/sp-switch.h
index 91fdcae..310655a 100644
--- a/src/sp-switch.h
+++ b/src/sp-switch.h
@@ -14,6 +14,7 @@
 
 #include "sp-item-group.h"
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 #define SP_TYPE_SWITCH            (CSwitch::getType())
diff --git a/src/sp-text.h b/src/sp-text.h
index d9b1539..b90fe37 100644
--- a/src/sp-text.h
+++ b/src/sp-text.h
@@ -14,6 +14,7 @@
  */
 
 #include <glib/gtypes.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "sp-item.h"
 #include "sp-string.h"
diff --git a/src/sp-tref-reference.h b/src/sp-tref-reference.h
index 37a5e21..8af5b53 100644
--- a/src/sp-tref-reference.h
+++ b/src/sp-tref-reference.h
@@ -13,6 +13,7 @@
 
 #include <forward.h>
 #include <uri-references.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 #include "util/share.h"
diff --git a/src/sp-use-reference.h b/src/sp-use-reference.h
index b30819a..46123e0 100644
--- a/src/sp-use-reference.h
+++ b/src/sp-use-reference.h
@@ -11,6 +11,7 @@
 
 #include <forward.h>
 #include <uri-references.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 class Path;
diff --git a/src/sp-use.h b/src/sp-use.h
index c5c0715..f886c07 100644
--- a/src/sp-use.h
+++ b/src/sp-use.h
@@ -13,6 +13,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "svg/svg-length.h"
 #include "sp-item.h"
diff --git a/src/spiral-context.h b/src/spiral-context.h
index d80c974..29a5f41 100644
--- a/src/spiral-context.h
+++ b/src/spiral-context.h
@@ -16,6 +16,7 @@
  */
 
 #include <gtk/gtktypeutils.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "event-context.h"
 #include "libnr/nr-point.h"
diff --git a/src/star-context.h b/src/star-context.h
index 024bf8d..3bc8ca3 100644
--- a/src/star-context.h
+++ b/src/star-context.h
@@ -14,6 +14,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "event-context.h"
 #include "libnr/nr-point.h"
diff --git a/src/style.h b/src/style.h
index f650cae..8589a48 100644
--- a/src/style.h
+++ b/src/style.h
@@ -24,6 +24,7 @@
 #include "uri.h"
 #include "sp-paint-server.h"
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 namespace Inkscape {
diff --git a/src/text-context.h b/src/text-context.h
index a6e2e8d..97501e8 100644
--- a/src/text-context.h
+++ b/src/text-context.h
@@ -15,6 +15,7 @@
  */
 
 /*  #include <gdk/gdkic.h> */
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <gtk/gtkimcontext.h>
 
diff --git a/src/ui/dialog/desktop-tracker.h b/src/ui/dialog/desktop-tracker.h
index 7a5bc39..b5b83b5 100644
--- a/src/ui/dialog/desktop-tracker.h
+++ b/src/ui/dialog/desktop-tracker.h
@@ -11,6 +11,7 @@
 #ifndef SEEN_DIALOG_DESKTOP_TRACKER
 #define SEEN_DIALOG_DESKTOP_TRACKER
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include <glib/gtypes.h>
 
diff --git a/src/ui/dialog/document-metadata.h b/src/ui/dialog/document-metadata.h
index f21bb0d..bb1dea7 100644
--- a/src/ui/dialog/document-metadata.h
+++ b/src/ui/dialog/document-metadata.h
@@ -14,6 +14,7 @@
 #define INKSCAPE_UI_DIALOG_DOCUMENT_METADATA_H
 
 #include <list>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <gtkmm/notebook.h>
 #include <glibmm/i18n.h>
diff --git a/src/ui/dialog/document-properties.h b/src/ui/dialog/document-properties.h
index c67dc97..08ba0ae 100644
--- a/src/ui/dialog/document-properties.h
+++ b/src/ui/dialog/document-properties.h
@@ -15,6 +15,7 @@
 #define INKSCAPE_UI_DIALOG_DOCUMENT_PREFERENCES_H
 
 #include <list>
+#include <stddef.h>
 #include <sigc++/sigc++.h>//
 #include <gtkmm/notebook.h>
 #include <glibmm/i18n.h>
diff --git a/src/ui/dialog/inkscape-preferences.h b/src/ui/dialog/inkscape-preferences.h
index 72975f7..31ec07a 100644
--- a/src/ui/dialog/inkscape-preferences.h
+++ b/src/ui/dialog/inkscape-preferences.h
@@ -28,6 +28,7 @@
 #include <gtkmm/frame.h>
 #include <gtkmm/notebook.h>
 #include "ui/widget/preferences-widget.h"
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <glibmm/i18n.h>
 #include <gtkmm/textview.h>
diff --git a/src/ui/dialog/undo-history.cpp b/src/ui/dialog/undo-history.cpp
index 8017af8..fa2eb00 100644
--- a/src/ui/dialog/undo-history.cpp
+++ b/src/ui/dialog/undo-history.cpp
@@ -14,6 +14,7 @@
 
 #include <glibmm/i18n.h>
 #include <gtk/gtkimage.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 
diff --git a/src/ui/tool/control-point-selection.h b/src/ui/tool/control-point-selection.h
index 8023c3e..f4e54e9 100644
--- a/src/ui/tool/control-point-selection.h
+++ b/src/ui/tool/control-point-selection.h
@@ -14,6 +14,7 @@
 
 #include <memory>
 #include <boost/optional.hpp>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <2geom/forward.h>
 #include <2geom/point.h>
diff --git a/src/ui/tool/control-point.h b/src/ui/tool/control-point.h
index 48c7074..ef5d046 100644
--- a/src/ui/tool/control-point.h
+++ b/src/ui/tool/control-point.h
@@ -12,6 +12,7 @@
 #define SEEN_UI_TOOL_CONTROL_POINT_H
 
 #include <boost/utility.hpp>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <gdkmm.h>
 #include <gtkmm.h>
diff --git a/src/ui/tool/manipulator.h b/src/ui/tool/manipulator.h
index 799dad0..f7be27d 100644
--- a/src/ui/tool/manipulator.h
+++ b/src/ui/tool/manipulator.h
@@ -13,6 +13,7 @@
 
 #include <set>
 #include <map>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <glib.h>
 #include <gdk/gdk.h>
diff --git a/src/ui/tool/multi-path-manipulator.h b/src/ui/tool/multi-path-manipulator.h
index 7a4cfdb..d7b8f8f 100644
--- a/src/ui/tool/multi-path-manipulator.h
+++ b/src/ui/tool/multi-path-manipulator.h
@@ -11,6 +11,7 @@
 #ifndef SEEN_UI_TOOL_MULTI_PATH_MANIPULATOR_H
 #define SEEN_UI_TOOL_MULTI_PATH_MANIPULATOR_H
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include "display/display-forward.h"
 #include "forward.h"
diff --git a/src/ui/tool/node-tool.h b/src/ui/tool/node-tool.h
index 39d04a1..f46b7dc 100644
--- a/src/ui/tool/node-tool.h
+++ b/src/ui/tool/node-tool.h
@@ -14,6 +14,7 @@
 #include <memory>
 #include <boost/ptr_container/ptr_map.hpp>
 #include <glib.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include "event-context.h"
 #include "forward.h"
diff --git a/src/ui/tool/node.h b/src/ui/tool/node.h
index 494bd35..563d094 100644
--- a/src/ui/tool/node.h
+++ b/src/ui/tool/node.h
@@ -15,6 +15,7 @@
 #include <iterator>
 #include <iosfwd>
 #include <stdexcept>
+#include <cstddef>
 #include <tr1/functional>
 #include <boost/utility.hpp>
 #include <boost/shared_ptr.hpp>
diff --git a/src/ui/view/view.h b/src/ui/view/view.h
index 882746c..e685355 100644
--- a/src/ui/view/view.h
+++ b/src/ui/view/view.h
@@ -15,6 +15,7 @@
  */
 
 #include <gdk/gdktypes.h>
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include "message.h"
 #include "gc-managed.h"
diff --git a/src/ui/widget/color-picker.h b/src/ui/widget/color-picker.h
index 477aa1c..167040c 100644
--- a/src/ui/widget/color-picker.h
+++ b/src/ui/widget/color-picker.h
@@ -13,6 +13,7 @@
 #ifndef __COLOR_PICKER_H__
 #define __COLOR_PICKER_H__
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <gtkmm/button.h>
 #include <gtkmm/dialog.h>
diff --git a/src/ui/widget/page-sizer.h b/src/ui/widget/page-sizer.h
index ba6c8dd..2072aec 100644
--- a/src/ui/widget/page-sizer.h
+++ b/src/ui/widget/page-sizer.h
@@ -13,6 +13,7 @@
 #define INKSCAPE_UI_WIDGET_PAGE_SIZER__H
 
 #include <gtkmm.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 #include "helper/units.h"
diff --git a/src/ui/widget/preferences-widget.h b/src/ui/widget/preferences-widget.h
index 6c45334..52e83e3 100644
--- a/src/ui/widget/preferences-widget.h
+++ b/src/ui/widget/preferences-widget.h
@@ -26,6 +26,7 @@
 #include <gtkmm/drawingarea.h>
 #include <gtkmm/frame.h>
 #include <gtkmm/filechooserbutton.h>
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 #include <glibmm/i18n.h>
 
diff --git a/src/ui/widget/selected-style.h b/src/ui/widget/selected-style.h
index 0229364..e74d5b1 100644
--- a/src/ui/widget/selected-style.h
+++ b/src/ui/widget/selected-style.h
@@ -23,6 +23,7 @@
 #include <gtkmm/adjustment.h>
 #include <gtkmm/spinbutton.h>
 
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 #include <glibmm/i18n.h>
diff --git a/src/ui/widget/style-subject.h b/src/ui/widget/style-subject.h
index 6f46eff..11c9ed0 100644
--- a/src/ui/widget/style-subject.h
+++ b/src/ui/widget/style-subject.h
@@ -14,6 +14,7 @@
 #include "libnr/nr-rect.h"
 #include <2geom/rect.h>
 #include "sp-item.h"
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 class SPDesktop;
diff --git a/src/uri-references.h b/src/uri-references.h
index a98c841..8d91481 100644
--- a/src/uri-references.h
+++ b/src/uri-references.h
@@ -13,6 +13,7 @@
  * Released under GNU GPL, read the file 'COPYING' for more information
  */
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 #include <sigc++/trackable.h>
 
diff --git a/src/util/forward-pointer-iterator.h b/src/util/forward-pointer-iterator.h
index 1603fed..0ee2460 100644
--- a/src/util/forward-pointer-iterator.h
+++ b/src/util/forward-pointer-iterator.h
@@ -15,6 +15,7 @@
 #define SEEN_INKSCAPE_UTIL_FORWARD_POINTER_ITERATOR_H
 
 #include <iterator>
+#include <cstddef>
 #include "util/reference.h"
 
 namespace Inkscape {
diff --git a/src/util/share.h b/src/util/share.h
index 3a2b735..74f3c3c 100644
--- a/src/util/share.h
+++ b/src/util/share.h
@@ -14,6 +14,7 @@
 
 #include "gc-core.h"
 #include <cstring>
+#include <cstddef>
 
 namespace Inkscape {
 namespace Util {
diff --git a/src/util/unordered-containers.h b/src/util/unordered-containers.h
index aaf7719..ea066c1 100644
--- a/src/util/unordered-containers.h
+++ b/src/util/unordered-containers.h
@@ -41,6 +41,8 @@
 # define INK_UNORDERED_MAP __gnu_cxx::hash_map
 # define INK_HASH __gnu_cxx::hash
 
+#include <cstddef>
+
 namespace __gnu_cxx {
 // hash function for pointers
 // TR1 and Boost have this defined by default, __gnu_cxx doesn't
diff --git a/src/widgets/desktop-widget.h b/src/widgets/desktop-widget.h
index 33f2a6a..0f71be1 100644
--- a/src/widgets/desktop-widget.h
+++ b/src/widgets/desktop-widget.h
@@ -21,6 +21,7 @@
 #include "ui/view/view-widget.h"
 #include "ui/view/edit-widget-interface.h"
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 // forward declaration
diff --git a/src/widgets/gradient-image.h b/src/widgets/gradient-image.h
index d0864b6..7b3854a 100644
--- a/src/widgets/gradient-image.h
+++ b/src/widgets/gradient-image.h
@@ -19,6 +19,7 @@ class SPGradient;
 
 #include <glib.h>
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 #define SP_TYPE_GRADIENT_IMAGE (sp_gradient_image_get_type ())
diff --git a/src/widgets/gradient-vector.h b/src/widgets/gradient-vector.h
index ceca915..808b3aa 100644
--- a/src/widgets/gradient-vector.h
+++ b/src/widgets/gradient-vector.h
@@ -17,6 +17,7 @@
 
 #include <glib.h>
 
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 #include <gtk/gtkvbox.h>
diff --git a/src/widgets/sp-attribute-widget.h b/src/widgets/sp-attribute-widget.h
index 01da29b..9940c4a 100644
--- a/src/widgets/sp-attribute-widget.h
+++ b/src/widgets/sp-attribute-widget.h
@@ -14,6 +14,7 @@
 #define SEEN_DIALOGS_SP_ATTRIBUTE_WIDGET_H
 
 #include <glib.h>
+#include <stddef.h>
 #include <sigc++/connection.h>
 
 #define SP_TYPE_ATTRIBUTE_WIDGET (sp_attribute_widget_get_type ())
diff --git a/src/widgets/sp-color-notebook.cpp b/src/widgets/sp-color-notebook.cpp
index 6c51130..8ddce8c 100644
--- a/src/widgets/sp-color-notebook.cpp
+++ b/src/widgets/sp-color-notebook.cpp
@@ -22,6 +22,7 @@
 #include <cstring>
 #include <string>
 #include <cstdlib>
+#include <cstddef>
 #include <gtk/gtk.h>
 #include <glibmm/i18n.h>
 
diff --git a/src/xml/helper-observer.h b/src/xml/helper-observer.h
index d028d39..e7881cd 100644
--- a/src/xml/helper-observer.h
+++ b/src/xml/helper-observer.h
@@ -5,6 +5,7 @@
 #include "node.h"
 #include "../sp-object.h"
 //#include "../sp-object-repr.h"
+#include <stddef.h>
 #include <sigc++/sigc++.h>
 
 namespace Inkscape {
-- 
1.7.3.4

